src/uart_connection.hpp

```diff
@@ -8,194 +8,193 @@
+     *
+     */
+    void begin();
+
+    /**
+     * @brief Check how many bytes are available to read.
+     *
+     * By checking the size of the receive buffer, we determine if new data has arrived.
+     *
+     * @return unsigned int Amount of bytes available to read.
+     */
+    unsigned int available();
+
+    /**
+     * @brief Enables the internal USART controller.
+     *
+     */
+    inline void enable();
+
+    /**
+     * @brief Disables the internal USART controller.
+     *
+     */
+    inline void disable();
+
+    /**
+     * @brief Send a single byte.
+     *
+     * @param c Byte.
+     * @return true Byte send.
+     * @return false Byte has not been send, USART controller not initialized.
+     */
+    bool send(const char c);
+
+    /**
+     * @brief Send a string.
+     *
+     * @param str String.
+     * @return true String send.
+     * @return false String has not been send, USART controller not initialized.
+     */
+    bool send(const char *str);
+
+    /**
+     * @brief Send a array of bytes with a specified length.
+     *
+     * @param data Array of bytes.
+     * @param length Length of array.
+     * @return true Array of bytes send.
+     * @return false Array of bytes has not been send, USART controller not initialized.
+     */
+    bool send(const char *data, size_t length);
+
+    /**
+     * @brief Receive a single byte.
+     *
+     * By popping the first element of the receive buffer, we return a received byte in a FIFO manner.
+     *
+     * @return char Received byte.
+     */
+    char receive();
+
+    /**
+     * @brief Checks if the internal USART controller has been initialized.
+     *
+     * @return true Internal USART controller is initialized.
+     * @return false Internal USART has not been initialized. See the begin() method.
+     */
+    bool isInitialized();
+
+    /**
+     * @brief Send a single byte.
+     *
+     * @param c Byte.
+     */
+    void operator<<(const char c);
+
+    /**
+     * @brief Send a string.
+     *
+     * @param str String (must be null terminated).
+     */
+    void operator<<(const char *str);
+
+    /**
+     * @brief Receive a byte.
+     *
+     * @param c Byte.
+     */
+    void operator>>(char &c);
+
+    /**
+     * @brief Destroy the UARTConnection object.
+     *
+     * Disables the UART controller to save resources.
+     *
+     */
+    ~UARTConnection();
+
+  private:
+    /**
+     * @brief Pointer the access the internal USART controllers.
+     *
+     */
+    Usart *hardwareUSART = nullptr;
+
+    /**
+     * @brief Data baudrate used for sending and receiving.
+     *
+     */
+    unsigned int baudrate;
+
+    /**
+     * @brief Enumerable type used to select on of the three USART controllers located on the Arduino Due.
+     *
+     */
+    UARTController controller;
+
+    /**
+     * @brief Holds the initialization status of the USART controller.
+     *
+     */
+    bool USARTControllerInitialized = false;
```

Comment:
Why not use the constructor to give it a value?

Reply:
Good point, but I think this should be addressed within the [UART LIB](https://github.com/R2D2-2018/UART_LIB) repository.